src/config.rs

```diff
@@ -526,61 +526,33 @@ impl<'a> Cfg<'a> {
         &self,
         force_install_active: Option<bool>,
     ) -> Result<Option<(LocalToolchainName, ActiveReason)>> {
-        let (components, targets, profile, toolchain, reason) = match self.find_override_config()? {
-            Some((
-                OverrideCfg::Official {
-                    components,
-                    targets,
-                    profile,
-                    toolchain,
-                },
-                reason,
-            )) => (components, targets, profile, toolchain, reason),
-            Some((override_config, reason)) => {
-                return Ok(Some((override_config.into_local_toolchain_name(), reason)));
-            }
-            None => {
-                return Ok(self
-                    .get_default()?
-                    .map(|x| (x.into(), ActiveReason::Default)));
-            }
-        };
-
         let should_install_active = if let Some(force) = force_install_active {
             force
         } else {
             self.should_auto_install()?
         };
-
         if !should_install_active {
-            return Ok(Some(((&toolchain).into(), reason)));
+            return self.active_toolchain();
```

Comment:
If `should_install_active` is set to `false`, we return `self.active_toolchain()` which was the old behavior of `find_active_toolchain()` in v1.28.0:

https://github.com/rust-lang/rustup/blob/e2d9e7edde6ee1a79e6e44f3b81065443c1fc6eb/src/config.rs#L517-L528